File: java/billing-service/src/main/java/net/multi/language/billing/service/PaymentService.java
package net.multi.language.billing.service;

import net.multi.language.billing.outbound.port.OutboundRestPort;
import net.multi.language.common.dto.order.OrderChargingStatusDTO;
import org.springframework.stereotype.Service;

@Service
public class PaymentService implements OutboundRestPort {

    // invokes a payment gateway (ex: Braintree) and returns the charging result
    @Override
    @SuppressWarnings("unused")
    public synchronized OrderChargingStatusDTO charge(final int paymentMethodId, final double amount) {
        return System.currentTimeMillis() % 2 == 0 ? new OrderChargingStatusDTO() :
                new OrderChargingStatusDTO(false, "The card has expired");
    }
}

File: python/src/billing_service/services/payment_service.py
from random import randint

from common.dto.order import OrderChargingStatusDTO


class PaymentService:
    # invokes a payment gateway (ex: Braintree) and returns the charging result
    def charge(self, payment_method_id: int, amount: float) -> OrderChargingStatusDTO:
        return OrderChargingStatusDTO(
            successful=bool(randint(0, 1)),
            failure_reason="The card has expired" if randint(0, 1) == 0 else None,
        )
